### PR TITLE
feat: at_app 5.0.0

### PR DESCRIPTION
**Pull request by xavierchanth**
**- What I did**

- Fixed melos config for melos 3.x
- Updated dependencies to support Dart 3
- Removed all package samples and --template flag from at_app
  - Associated changes required in the code and CI
- Removed the locally run build scripts
  - Moved them to only be available as composite actions
- Fixed analysis warnings in snackbar/snackbar_sender demos

**- How I did it**

**- How to verify it**

**- Description for the changelog**
feat: at_app 5.0.0

### CODE REVIEW

**Review by cpswan — DISMISSED** [review]

**Comment by xavierchanth** [comment]
@cpswan this was supposed to be a PR to run my changes against the CI, but I'm realizing now that I won't be able to stack these changes against my other PR... due to some nasty merge conflicts that probably only make this series of updates more complicated.

Let's review this together when we can, even though there is a lot of changes, it should be very quick because a majority of the lines here are the removal of the sample templates and associated work.

**Comment by xavierchanth** [comment]
Ah Semantic PR now updates its status upon a rename of the PR :)

File: templates/demos/snackbar_sender/__brick__/lib/main.dart [hunk 1]
```diff
@@ -38,6 +38,7 @@ class _MyAppState extends State<MyApp> {
   Future<AtClientPreference> futurePreference = loadAtClientPreference();
   AtClientPreference? atClientPreference;
 
+  // ignore: unused_field
```
Comment (xavierchanth) [inline templates/demos/snackbar_sender/__brick__/lib/main.dart:42]: // _logger is left in place for anyone wishing to extend the example

File: templates/demos/snackbar_sender/pubspec.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 name: snackbar_sender_template
 publish_to: none
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: ">=2.12.0 <3.0.0"
```
Comment (xavierchanth) [inline templates/demos/snackbar_sender/pubspec.yaml]: 4.0.0

**Comment by xavierchanth** [comment]
@cpswan waiting for the CI to finish:
https://github.com/atsign-foundation/at_app/actions/runs/5135462686

**Review by cpswan — APPROVED** [review]

**Pull request merged**